use bestBidPrice and bestAskPrice to calculate probability of trading in liquidity

jeremyletang opened the pull request:
close #3424

jeremyletang committed:
use bestBidPrice and bestAskPrice to calculate probability of trading in liquidity

ashleyvega reviewed (changes requested):
Calls to `fmt.Printf()` should be removed.

jeremyletang committed:
update lp feature tests to work with the getProbabilityOfTrading changes

use .5 prob of trading as default for lp order prie in range of bestBid and BestAsk

implement minProbabilityOfTradingForLPOrder netparam

jeremyletang committed:
ensure all price are in the minPrice/maxPrice range when price liquidity order

jeremyletang committed:
returns the pegged order when calling repricePeggedFN + use it in order

ValentinTrinque committed:
Do not check liquidity provision submission when commitment amount is 0

witgaw committed:
Add comment

witgaw committed:
Change Ceil->Floor (in line with comment)

witgaw committed:
Fix tests

Note: other component now assures that provided prices are always within
within the valid price rage, hence checks on invalid price cases have
become redundant.

witgaw committed:
Amend comment

jeremyletang committed:
remove all tests debugs

jeremyletang committed:
fix typos

jeremyletang committed:
remove debug

jeremyletang committed:
fix liquidity provision cancel
